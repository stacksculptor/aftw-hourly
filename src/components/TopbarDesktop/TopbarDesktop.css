@import '../../marketplace.css';

:root {
  /* Compose text label items on top of this */
  /* This stretches inline element (link) to take available vertical space (big hover area),
   * and align baselines
   */
  --topbarDesktopLabelStyles: {
    display: inline-block;
    margin: 24px 0;
    text-decoration: inherit;

    @media (--desktopViewport) {
      margin: 24px 0;
    }
  }

  --linkHoverState: {
    border-bottom: 0px solid;
    transition: var(--transitionStyleButton);

    &:hover {
      border-bottom: 4px solid var(--marketplaceColor);
      text-decoration: none;
    }
  }
}

/* Desktop */
.root {
  /* Size */
  width: 100%;
  height: var(--topbarHeightDesktop);

  /* Layout for child components */
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: flex-start;
  padding: 0 24px 0 0;

  /* fill */
  background-color: var(--matterColorLight);

  /* shadows */
  box-shadow: var(--boxShadowLight);
}

/* logo */
.logoLink {
  @apply --linkHoverState;
  flex-shrink: 0;
  padding: 21px 24px 21px 24px;
  height: 100%;

  @media (min-width: 1024px) {
    padding: 21px 36px 21px 36px;
  }
}

.logo {
  width: 155px;
  height: 27px;
}

/* Search */
.searchLink {
  flex-grow: 1;
  height: 100%;
  margin-right: 24px;
  border-left-style: solid;
  border-left-width: 1px;
  border-left-color: var(--matterColorNegative);

  @media (min-width: 1024px) {
    padding-left: 0;
  }
}

.search {
  @apply --marketplaceH4FontStyles;
  @apply --topbarDesktopLabelStyles;
  color: var(--matterColor);
}

.topbarSearchWithLeftPadding {
  padding-left: 24px;
  height: var(--topbarHeightDesktop);

  @media (min-width: 1024px) {
    padding-left: 34px;
  }
}

/* Create listing (CTA for providers) */
.createListingLink {
  @apply --linkHoverState;
  @apply --marketplaceH4FontStyles;

  flex-shrink: 0;
  height: 100%;
  padding: 0 12px 0 12px;
  color: var(--marketplaceColor);
  margin-top: 0;
  margin-bottom: 0;

  &:hover {
    color: var(--marketplaceColorDark);
  }

  @media (--desktopViewport) {
   margin-top: 0;
   margin-bottom: 0;
  }
}

.createListing {
  @apply --topbarDesktopLabelStyles;
}

/* Inbox */
.inboxLink {
  @apply --linkHoverState;
  @apply --marketplaceH4FontStyles;
  height: 100%;
  padding: 0 12px 0 12px;
  margin-top: 0;
  margin-bottom: 0;
  color: var(--matterColor);

  &:hover {
    color: var(--matterColorDark);
  }

  @media (--desktopViewport) {
   margin-top: 0;
   margin-bottom: 0;
  }
}

.inbox {
  @apply --topbarDesktopLabelStyles;
  position: relative;
}

.notificationDot {
  /* Dimensions */
  width: 7px;
  height: 7px;
  border-radius: 50%;

  /* Position */
  position: absolute;
  top: 3px;
  right: -9px;

  /* Style: red dot */
  background-color: var(--failColor);
}

/* Profile menu */
.profileMenuLabel {
  @apply --linkHoverState;
  flex-shrink: 0;
  display: flex;
  align-items: center;
  height: 100%;
  margin: 0;
  padding: 0 12px 0 12px;

  &:active {
    border-bottom: 0;
  }
}

.avatar {
  margin: 16px 0;
  width: 41px;
  height: 41px;
}

.profileMenuContent {
  min-width: 276px;
}

/* Authentication */
.signupLink {
  @apply --linkHoverState;
  flex-shrink: 0;
  height: 100%;
  padding: 0 12px 0 12px;

  color: var(--matterColor);

  &:hover {
    color: var(--matterColorDark);
  }
}

.loginLink {
  @apply --linkHoverState;
  flex-shrink: 0;
  height: 100%;
  padding: 0 12px 0 12px;

  color: var(--matterColor);

  &:hover {
    color: var(--matterColorDark);
  }
}

.signup,
.login {
  @apply --marketplaceH4FontStyles;
  @apply --topbarDesktopLabelStyles;
}

.logoutButton {
  @apply --marketplaceH4FontStyles;
  width: 100%;
  min-width: 276px;
  margin: 0;
  padding: 20px 24px;
  color: var(--matterColor);
  text-align: left;
  white-space: nowrap;
  transition: var(--transitionStyleButton);

  &:hover {
    color: var(--matterColorDark);
    text-decoration: none;
    border-left: 6px solid var(--marketplaceColor);
  }

  @media (--desktopViewport) {
    margin: 0;
  }

  &:focus {
    outline: none;
  }
}
