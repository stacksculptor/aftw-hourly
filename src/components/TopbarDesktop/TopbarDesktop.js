import React, { PropTypes } from 'react';
import { FormattedMessage, intlShape } from 'react-intl';
import classNames from 'classnames';
import {
  Avatar,
  InlineTextButton,
  Menu,
  MenuLabel,
  MenuContent,
  MenuItem,
  NamedLink,
} from '../../components';
import { TopbarSearchForm } from '../../containers';

import logo from './images/saunatime-logo.png';
import css from './TopbarDesktop.css';

const TopbarDesktop = props => {
  const {
    className,
    firstName,
    lastName,
    rootClassName,
    currentUserHasListings,
    notificationCount,
    intl,
    isAuthenticated,
    onLogout,
    onSearchSubmit,
    initialSearchFormValues,
  } = props;

  const rootClass = rootClassName || css.root;
  const classes = classNames(rootClass, className);

  const search = (
    <TopbarSearchForm
      className={css.searchLink}
      desktopInputRoot={css.topbarSearchWithLeftPadding}
      form="TopbarSearchFormDesktop"
      onSubmit={onSearchSubmit}
      initialValues={initialSearchFormValues}
    />
  );

  const notificationDot = notificationCount > 0 ? <div className={css.notificationDot} /> : null;

  const inboxLink = isAuthenticated
    ? <NamedLink
        className={css.inboxLink}
        name="InboxPage"
        params={{ tab: currentUserHasListings ? 'sales' : 'orders' }}
      >
        <span className={css.inbox}>
          <FormattedMessage id="TopbarDesktop.inbox" />
          {notificationDot}
        </span>
      </NamedLink>
    : null;

  const profileMenu = isAuthenticated
    ? <Menu>
        <MenuLabel className={css.profileMenuLabel} isOpenClassName={css.profileMenuIsOpen}>
          <Avatar className={css.avatar} firstName={firstName} lastName={lastName} />
        </MenuLabel>
        <MenuContent className={css.profileMenuContent}>
          <MenuItem key="logout">
            <InlineTextButton className={css.logoutButton} onClick={onLogout}>
              <FormattedMessage id="TopbarDesktop.logout" />
            </InlineTextButton>
          </MenuItem>
        </MenuContent>
      </Menu>
    : null;

  const signupLink = isAuthenticated
    ? null
    : <NamedLink name="SignupPage" className={css.signupLink}>
        <span className={css.signup}><FormattedMessage id="TopbarDesktop.signup" /></span>
      </NamedLink>;

  const loginLink = isAuthenticated
    ? null
    : <NamedLink name="LoginPage" className={css.loginLink}>
        <span className={css.login}><FormattedMessage id="TopbarDesktop.login" /></span>
      </NamedLink>;

  return (
    <nav className={classes}>
      <NamedLink className={css.logoLink} name="LandingPage">
        <img
          className={css.logo}
          src={logo}
          alt={intl.formatMessage({ id: 'TopbarDesktop.logo' })}
        />
      </NamedLink>
      {search}
      <NamedLink className={css.createListingLink} name="NewListingPage">
        <span className={css.createListing}>
          <FormattedMessage id="TopbarDesktop.createListing" />
        </span>
      </NamedLink>
      {inboxLink}
      {profileMenu}
      {signupLink}
      {loginLink}
    </nav>
  );
};

const { bool, func, object, number, string } = PropTypes;

TopbarDesktop.defaultProps = {
  firstName: '',
  lastName: '',
  notificationCount: 0,
  className: null,
  rootClassName: null,
  initialSearchFormValues: {},
};

TopbarDesktop.propTypes = {
  className: string,
  currentUserHasListings: bool.isRequired,
  isAuthenticated: bool.isRequired,
  onLogout: func.isRequired,
  firstName: string,
  lastName: string,
  notificationCount: number,
  rootClassName: string,
  onSearchSubmit: func.isRequired,
  initialSearchFormValues: object,
  intl: intlShape.isRequired,
};

export default TopbarDesktop;
