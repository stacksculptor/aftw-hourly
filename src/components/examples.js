/* eslint-disable import/prefer-default-export*/
import * as Hello from './Hello/Hello.example';

export { Hello };
