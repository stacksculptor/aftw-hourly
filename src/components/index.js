import AddImages from './AddImages/AddImages';
import AuthorInfo from './AuthorInfo/AuthorInfo';
import BookingInfo from './BookingInfo/BookingInfo';
import Button from './Button/Button';
import CurrencyInput from './CurrencyInput/CurrencyInput';
import DateInput from './DateInput/DateInput';
import Discussion from './Discussion/Discussion';
import FilterPanel from './FilterPanel/FilterPanel';
import HeroSection from './HeroSection/HeroSection';
import Input from './Input/Input';
import LabeledField from './LabeledField/LabeledField';
import ListingCard from './ListingCard/ListingCard';
import ListingCardSmall from './ListingCardSmall/ListingCardSmall';
import LocationAutocompleteInput from './LocationAutocompleteInput/LocationAutocompleteInput';
import Map from './Map/Map';
import MapPanel from './MapPanel/MapPanel';
import Menu from './Menu/Menu';
import MobileFrame from './MobileFrame/MobileFrame';
import ModalInMobile from './ModalInMobile/ModalInMobile';
import NamedLink from './NamedLink/NamedLink';
import NamedRedirect from './NamedRedirect/NamedRedirect';
import OrderDetailsPanel from './OrderDetailsPanel/OrderDetailsPanel';
import OrderDiscussionPanel from './OrderDiscussionPanel/OrderDiscussionPanel';
import PageLayout from './PageLayout/PageLayout';
import PaginationLinks from './PaginationLinks/PaginationLinks';
import Promised from './Promised/Promised';
import RoutesProvider from './RoutesProvider/RoutesProvider';
import SearchResultsPanel from './SearchResultsPanel/SearchResultsPanel';
import StripeBankAccountToken from './StripeBankAccountToken/StripeBankAccountToken';

export {
  AddImages,
  AuthorInfo,
  BookingInfo,
  Button,
  CurrencyInput,
  DateInput,
  Discussion,
  FilterPanel,
  HeroSection,
  Input,
  LabeledField,
  ListingCard,
  ListingCardSmall,
  LocationAutocompleteInput,
  Map,
  MapPanel,
  Menu,
  MobileFrame,
  ModalInMobile,
  NamedLink,
  NamedRedirect,
  OrderDetailsPanel,
  OrderDiscussionPanel,
  PageLayout,
  PaginationLinks,
  Promised,
  RoutesProvider,
  SearchResultsPanel,
  StripeBankAccountToken,
};
