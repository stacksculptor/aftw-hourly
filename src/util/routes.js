import React, { PropTypes } from 'react';
import { find } from 'lodash';
import { matchPath } from 'react-router-dom';
import pathToRegexp from 'path-to-regexp';
import { stringify } from './urlHelpers';
import * as propTypes from './propTypes';

// Flatten the routes config.
// TODO: flatten the original config and remove this function
export const flattenRoutes = routes =>
  routes.reduce(
    (flatRoutes, route) => {
      const r = { ...route };
      delete r.routes;
      flatRoutes.push(r);
      if (route.routes) {
        return flatRoutes.concat(flattenRoutes(route.routes));
      }
      return flatRoutes;
    },
    []
  );

const findRouteByName = (nameToFind, flattenedRoutes) =>
  find(flattenedRoutes, route => route.name === nameToFind);

/**
 * E.g. ```const toListingPath = toPathByRouteName('ListingPage', routes);```
 * Then we can generate listing paths with given params (```toListingPath({ id: uuidX })```)
 */
const toPathByRouteName = (nameToFind, flattenedRoutes) => {
  const route = findRouteByName(nameToFind, flattenedRoutes);
  if (!route) {
    throw new Error(`Path "${nameToFind}" was not found.`);
  }
  return pathToRegexp.compile(route.path);
};

/**
 * Shorthand for single path call. (```pathByRouteName('ListingPage', routes, { id: uuidX });```)
 */
export const pathByRouteName = (nameToFind, flattenedRoutes, params = {}) =>
  toPathByRouteName(nameToFind, flattenedRoutes)(params);

/**
 * Find the matching routes and their params for the given pathname
 *
 * @param {String} pathname - Full URL path from root with possible
 * search params and hash included
 *
 * @return {Array<{ route, params }>} - All matches as { route, params } objects
 */
export const matchPathname = (pathname, routesConfiguration) => {
  // TODO: remove flattening when routesConfiguration is flat
  const flattenedRoutes = flattenRoutes(routesConfiguration);

  return flattenedRoutes.reduce(
    (matches, route) => {
      const { path, exact = false } = route;
      const match = matchPath(pathname, { path, exact });
      if (match) {
        matches.push({
          route,
          params: match.params || {},
        });
      }
      return matches;
    },
    []
  );
};

/**
 * A higher order component (HOC) to take the flattened routes from
 * the context that the RoutesProvider component has provided.
 *
 * Injects the routes as the `flattenedRoutes` prop in the given
 * component. Works similarly as `withRouter` in React Router.
 */
export const withFlattenedRoutes = Component => {
  const WithFlattenedRoutesComponent = (props, context) => (
    <Component flattenedRoutes={context.flattenedRoutes} {...props} />
  );

  WithFlattenedRoutesComponent.displayName = `withFlattenedRoutes(${Component.displayName || Component.name})`;

  const { arrayOf } = PropTypes;

  WithFlattenedRoutesComponent.contextTypes = {
    flattenedRoutes: arrayOf(propTypes.route).isRequired,
  };

  return WithFlattenedRoutesComponent;
};

/**
 * ResourceLocatorString is used to direct webapp to correct page.
 * In contrast to Universal Resource Locator (URL), this doesn't contain protocol, host, or port.
 */
export const createResourceLocatorString = (
  routeName,
  flattenedRoutes,
  pathParams = {},
  searchParams = {}
) => {
  const searchQuery = stringify(searchParams);
  const includeSearchQuery = searchQuery.length > 0 ? `?${searchQuery}` : '';
  const path = pathByRouteName(routeName, flattenedRoutes, pathParams);
  return `${path}${includeSearchQuery}`;
};
