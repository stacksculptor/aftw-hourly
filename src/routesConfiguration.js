import React from 'react';
import { NamedRedirect } from './components';
import {
  AuthenticationPage,
  CheckoutPage,
  ContactDetailsPage,
  EditListingPage,
  InboxPage,
  LandingPage,
  ListingPage,
  ManageListingsPage,
  NotFoundPage,
  OrderPage,
  PasswordForgottenPage,
  PasswordResetPage,
  PayoutPreferencesPage,
  ProfilePage,
  ProfileSettingsPage,
  SalePage,
  SearchPage,
  SecurityPage,
  StyleguidePage,
  EmailVerificationPage,
} from './containers';

// https://en.wikipedia.org/wiki/Universally_unique_identifier#Nil_UUID
const draftId = '00000000-0000-0000-0000-000000000000';
const draftSlug = 'draft';

const RedirectToLandingPage = () => <NamedRedirect name="LandingPage" />;

const routesConfiguration = [
  { path: '/', exact: true, name: 'LandingPage', component: props => <LandingPage {...props} /> },
  {
    path: '/s',
    exact: true,
    name: 'SearchPage',
    component: props => <SearchPage {...props} />,
    loadData: (params, search) => SearchPage.loadData(params, search),
    routes: [
      {
        path: '/s/filters',
        exact: true,
        name: 'SearchFiltersPage',
        component: props => <SearchPage {...props} tab="filters" />,
        loadData: (params, search) => SearchPage.loadData(params, search),
      },
      {
        path: '/s/listings',
        exact: true,
        name: 'SearchListingsPage',
        component: props => <SearchPage {...props} tab="listings" />,
        loadData: (params, search) => SearchPage.loadData(params, search),
      },
      {
        path: '/s/map',
        exact: true,
        name: 'SearchMapPage',
        component: props => <SearchPage {...props} tab="map" />,
        loadData: (params, search) => SearchPage.loadData(params, search),
      },
    ],
  },
  {
    path: '/l',
    exact: true,
    name: 'ListingBasePage',
    component: RedirectToLandingPage,
    routes: [
      {
        path: '/l/:slug/:id',
        exact: true,
        name: 'ListingPage',
        loadData: (params, search) => ListingPage.loadData(params, search),
        component: props => <ListingPage {...props} tab="listing" />,
      },
      {
        path: '/l/:slug/:id/book',
        exact: true,
        name: 'ListingPage',
        loadData: (params, search) => ListingPage.loadData(params, search),
        component: props => <ListingPage {...props} tab="book" />,
      },
      {
        path: '/l/:slug/:id/checkout',
        auth: true,
        exact: true,
        name: 'CheckoutPage',
        setInitialValues: initialValues => CheckoutPage.setInitialValues(initialValues),
        component: props => <CheckoutPage {...props} />,
      },
      {
        auth: true,
        path: '/l/new',
        exact: true,
        name: 'NewListingPage',
        component: () => (
          <NamedRedirect
            name="EditListingPage"
            params={{ slug: draftSlug, id: draftId, type: 'new', tab: 'description' }}
          />
        ),
      },
      {
        auth: true,
        path: '/l/:slug/:id/:type/:tab',
        exact: true,
        name: 'EditListingPage',
        component: props => <EditListingPage {...props} />,
        loadData: (params, search) => EditListingPage.loadData(params, search),
      },
    ],
  },
  {
    path: '/u',
    exact: true,
    name: 'ProfileBasePage',
    component: RedirectToLandingPage,
    routes: [
      {
        path: '/u/:displayName',
        exact: true,
        name: 'ProfilePage',
        component: props => <ProfilePage {...props} />,
      },
    ],
  },
  {
    path: '/profile-settings',
    auth: true,
    exact: true,
    name: 'ProfileSettingsPage',
    component: props => <ProfileSettingsPage {...props} />,
  },
  {
    path: '/login',
    exact: true,
    name: 'LoginPage',
    component: props => <AuthenticationPage {...props} tab="login" />,
  },
  {
    path: '/signup',
    exact: true,
    name: 'SignupPage',
    component: props => <AuthenticationPage {...props} tab="signup" />,
  },
  {
    path: '/password',
    exact: true,
    name: 'PasswordPage',
    component: props => <PasswordForgottenPage {...props} />,
  },
  {
    path: '/password/forgotten',
    exact: true,
    name: 'PasswordForgottenPage',
    component: props => <PasswordForgottenPage {...props} />,
  },
  {
    path: '/inbox',
    auth: true,
    exact: true,
    name: 'InboxBasePage',
    component: () => <NamedRedirect name="InboxPage" params={{ tab: 'sales' }} />,
  },
  {
    path: '/inbox/:tab',
    auth: true,
    exact: true,
    name: 'InboxPage',
    component: props => <InboxPage {...props} />,
    loadData: (params, search) => InboxPage.loadData(params, search),
  },
  {
    path: '/order/:id',
    auth: true,
    exact: true,
    name: 'OrderPage',
    component: RedirectToLandingPage,
    routes: [
      {
        path: '/order/:id/details',
        auth: true,
        exact: true,
        name: 'OrderDetailsPage',
        component: props => <OrderPage {...props} tab="details" />,
        loadData: params => OrderPage.loadData(params),
      },
      {
        path: '/order/:id/discussion',
        auth: true,
        exact: true,
        name: 'OrderDiscussionPage',
        component: props => <OrderPage {...props} tab="discussion" />,
        loadData: params => OrderPage.loadData(params),
      },
    ],
  },
  {
    path: '/sale/:id',
    auth: true,
    exact: true,
    name: 'SalePage',
    component: props => <SalePage {...props} tab="discussion" />,
    routes: [
      {
        path: '/sale/:id/details',
        auth: true,
        exact: true,
        name: 'SaleDetailsPage',
        component: props => <SalePage {...props} tab="details" />,
        loadData: params => SalePage.loadData(params),
      },
      {
        path: '/sale/:id/discussion',
        auth: true,
        exact: true,
        name: 'SaleDiscussionPage',
        component: props => <SalePage {...props} tab="discussion" />,
        loadData: params => SalePage.loadData(params),
      },
    ],
  },
  {
    path: '/listings',
    auth: true,
    exact: true,
    name: 'ManageListingsPage',
    component: props => <ManageListingsPage {...props} />,
    loadData: (params, search) => ManageListingsPage.loadData(params, search),
  },
  {
    path: '/account',
    auth: true,
    exact: true,
    name: 'AccountPage',
    component: () => <NamedRedirect name="ContactDetailsPage" />,
    routes: [
      {
        path: '/account/contact-details',
        auth: true,
        exact: true,
        name: 'ContactDetailsPage',
        component: props => <ContactDetailsPage {...props} />,
      },
      {
        path: '/account/payout-preferences',
        auth: true,
        exact: true,
        name: 'PayoutPreferencesPage',
        component: props => <PayoutPreferencesPage {...props} />,
      },
      {
        path: '/account/security',
        auth: true,
        exact: true,
        name: 'SecurityPage',
        component: props => <SecurityPage {...props} />,
      },
    ],
  },
  {
    path: '/styleguide',
    exact: true,
    name: 'Styleguide',
    component: props => <StyleguidePage {...props} />,
  },
  {
    path: '/styleguide/g/:group',
    exact: true,
    name: 'StyleguideGroup',
    component: props => <StyleguidePage {...props} />,
  },
  {
    path: '/styleguide/c/:component',
    exact: true,
    name: 'StyleguideComponent',
    component: props => <StyleguidePage {...props} />,
  },
  {
    path: '/styleguide/c/:component/:example',
    exact: true,
    name: 'StyleguideComponentExample',
    component: props => <StyleguidePage {...props} />,
  },
  {
    path: '/styleguide/c/:component/:example/raw',
    exact: true,
    name: 'StyleguideComponentExampleRaw',
    component: props => <StyleguidePage raw {...props} />,
  },
  {
    path: '/notfound',
    exact: true,
    name: 'NotFoundPage',
    component: props => <NotFoundPage {...props} />,
  },

  // Do not change this path!
  //
  // The API expects that the Starter App implements /reset-password endpoint
  {
    path: '/reset-password',
    exact: true,
    name: 'PasswordResetPage',
    component: props => <PasswordResetPage {...props} />,
  },

  // Do not change this path!
  //
  // The API expects that the Starter App implements /verify-email endpoint
  {
    path: '/verify-email',
    auth: true,
    authPage: 'LoginPage',
    exact: true,
    name: 'EmailVerificationPage',
    component: props => <EmailVerificationPage {...props} />,
  },
];

export default routesConfiguration;
