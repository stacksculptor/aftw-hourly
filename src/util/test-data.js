import Decimal from 'decimal.js';
import { types } from './sdkLoader';
import { nightsBetween } from '../util/dates';
import * as propTypes from '../util/propTypes';

const { UUID, LatLng, Money } = types;

// Create a booking that conforms to the util/propTypes booking schema
export const createBooking = (id, attributes = {}) => ({
  id: new UUID(id),
  type: 'booking',
  attributes: {
    start: new Date(Date.UTC(2017, 5, 10)),
    end: new Date(Date.UTC(2017, 5, 10)),
    ...attributes,
  },
});

// Create a user that conforms to the util/propTypes user schema
export const createUser = id => ({
  id: new UUID(id),
  type: 'user',
  attributes: {
    banned: false,
    profile: {
      displayName: `${id} display name`,
      abbreviatedName: `${id} abbreviated name`,
    },
  },
});

// Create a user that conforms to the util/propTypes currentUser schema
export const createCurrentUser = id => ({
  id: new UUID(id),
  type: 'current-user',
  attributes: {
    banned: false,
    email: `${id}@example.com`,
    emailVerified: true,
    profile: {
      firstName: `${id} first name`,
      lastName: `${id} last name`,
      displayName: `${id} display name`,
      abbreviatedName: `${id} abbreviated name`,
    },
    stripeConnected: true,
  },
});

// Create a user that conforms to the util/propTypes user schema
export const createImage = id => ({
  id: new UUID(id),
  type: 'image',
  attributes: {
    sizes: [
      {
        name: 'square',
        height: 408,
        width: 408,
        url: 'https://placehold.it/408x408',
      },
      {
        name: 'square2x',
        height: 816,
        width: 816,
        url: 'https://placehold.it/816x816',
      },
    ],
  },
});

// Create a user that conforms to the util/propTypes listing schema
export const createListing = (id, attributes = {}, includes = {}) => ({
  id: new UUID(id),
  type: 'listing',
  attributes: {
    title: `${id} title`,
    description: `${id} description`,
    address: `${id} address`,
    geolocation: new LatLng(40, 60),
    closed: false,
    deleted: false,
    price: new Money(5500, 'USD'),
    ...attributes,
  },
  ...includes,
});

export const createTransaction = options => {
  const {
    id,
    lastTransition = propTypes.TX_TRANSITION_PREAUTHORIZE,
    total = new Money(1000, 'USD'),
    commission = new Money(100, 'USD'),
    booking = null,
    listing = null,
    customer = null,
    provider = null,
    lastTransitionedAt = new Date(Date.UTC(2017, 5, 1)),
  } = options;
  const nightCount = booking ? nightsBetween(booking.attributes.start, booking.attributes.end) : 1;
  return {
    id: new UUID(id),
    type: 'transaction',
    attributes: {
      createdAt: new Date(Date.UTC(2017, 4, 1)),
      lastTransitionedAt,
      lastTransition,
      payinTotal: total,
      payoutTotal: new Money(total.amount - commission.amount, total.currency),
      lineItems: [
        {
          code: 'line-item/night',
          quantity: new Decimal(nightCount),
          unitPrice: new Money(total.amount / nightCount, total.currency),
          lineTotal: total,
          reversal: false,
        },
        {
          code: 'line-item/provider-commission',
          unitPrice: new Money(commission.amount * -1, commission.currency),
          lineTotal: new Money(commission.amount * -1, commission.currency),
          reversal: false,
        },
      ],
    },
    booking,
    listing,
    customer,
    provider,
  };
};

// Default config for currency formatting in tests and examples.
export const currencyConfig = {
  style: 'currency',
  currency: 'USD',
  currencyDisplay: 'symbol',
  useGrouping: true,
  minimumFractionDigits: 2,
  maximumFractionDigits: 2,
};

const pad = num => {
  return num >= 0 && num < 10 ? `0${num}` : `${num}`;
};

// Create fake Internalization object to help with shallow rendering.
export const fakeIntl = {
  formatDate: d => `${d.getUTCFullYear()}-${pad(d.getUTCMonth() + 1)}-${pad(d.getUTCDate())}`,
  formatHTMLMessage: d => d,
  formatMessage: msg => msg.id,
  formatNumber: d => d,
  formatPlural: d => d,
  formatRelative: d => d,
  formatTime: d => `${pad(d.getUTCHours())}:${pad(d.getUTCMinutes())}`,
  now: () => Date.UTC(2017, 10, 23, 12, 59),
};

const noop = () => null;

export const fakeFormProps = {
  anyTouched: false,
  asyncValidating: false,
  dirty: false,
  form: 'fakeTestForm',
  invalid: false,
  pristine: true,
  clearSubmit: noop,
  touch: noop,
  untouch: noop,
  submit: noop,
  reset: noop,
  initialize: noop,
  handleSubmit: noop,
  destroy: noop,
  clearAsyncError: noop,
  change: noop,
  blur: noop,
  autofill: noop,
  asyncValidate: noop,
  valid: true,
  submitSucceeded: false,
  submitFailed: false,
  submitting: false,
  pure: true,
  initialized: true,
};
