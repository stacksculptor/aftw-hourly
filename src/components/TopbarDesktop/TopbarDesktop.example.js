/* eslint-disable no-console */
import { fakeIntl } from '../../util/test-data';
import TopbarDesktop from './TopbarDesktop';

const noop = () => null;

export const AuthenticatedDesktopTopbar = {
  component: TopbarDesktop,
  props: {
    isAuthenticated: true,
    currentUserHasListings: true,
    name: 'John Doe',
    onSearchSubmit: values => {
      console.log('submit search:', values);
    },
    intl: fakeIntl,
    onLogout: noop,
  },
  group: 'navigation',
};
