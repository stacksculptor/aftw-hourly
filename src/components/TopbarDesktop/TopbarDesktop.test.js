import React from 'react';
import { fakeIntl } from '../../util/test-data';
import { renderDeep } from '../../util/test-helpers';
import { RoutesProvider } from '../../components';
import routesConfiguration from '../../routesConfiguration';
import { flattenRoutes } from '../../util/routes';
import TopbarDesktop from './TopbarDesktop';

const noop = () => null;

describe('TopbarDesktop', () => {
  it('data matches snapshot', () => {
    const flattenedRoutes = flattenRoutes(routesConfiguration);
    const tree = renderDeep(
      <RoutesProvider flattenedRoutes={flattenedRoutes}>
        <TopbarDesktop
          isAuthenticated
          currentUserHasListings
          name="John Doe"
          intl={fakeIntl}
          onLogout={noop}
        />
      </RoutesProvider>
    );
    expect(tree).toMatchSnapshot();
  });
});
