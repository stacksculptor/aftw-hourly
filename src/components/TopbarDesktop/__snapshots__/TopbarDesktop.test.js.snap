exports[`TopbarDesktop data matches snapshot 1`] = `
<nav
  className="">
  <a
    className="NamedLink_active"
    href="/"
    onClick={[Function]}
    style={Object {}}>
    <img
      alt="TopbarDesktop.logo"
      className={undefined}
      src="saunatime-logo.png" />
  </a>
  <form
    className=""
    onSubmit={[Function]}>
    <div
      className="">
      <div
        className="">
        <svg
          className={undefined}
          height="22"
          viewBox="0 0 21 22"
          width="21"
          xmlns="http://www.w3.org/2000/svg">
          <g
            className={undefined}
            fill="none"
            fillRule="evenodd"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            transform="matrix(-1 0 0 1 20 1)">
            <path
              d="M13 14l5.241 5.241" />
            <circle
              cx="7.5"
              cy="7.5"
              r="7.5" />
          </g>
        </svg>
      </div>
      <input
        autoComplete="off"
        autoFocus={false}
        className=""
        name="location"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Search saunas..."
        type="search"
        value="" />
    </div>
  </form>
  <a
    className=""
    href="/l/new"
    onClick={[Function]}
    style={Object {}}>
    <span
      className={undefined}>
      <span>
        + Add your sauna
      </span>
    </span>
  </a>
  <a
    className=""
    href="/inbox/sales"
    onClick={[Function]}
    style={Object {}}>
    <span
      className={undefined}>
      <span>
        Inbox
      </span>
    </span>
  </a>
  <div
    className=""
    onBlur={[Function]}
    onKeyDown={[Function]}>
    <button
      className=""
      onClick={[Function]}>
      <div
        className=""
        title="">
        <span>
          
        </span>
      </div>
    </button>
    <div
      className=""
      style={Object {}}>
      <ul
        className="">
        <li
          className=""
          role="menuitem">
          <button
            className=""
            onClick={[Function]}>
            <span>
              Log out
            </span>
          </button>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
